src/common/utils/pagination.ts

```diff
@@ -0,0 +1,43 @@
+import { DeserializedList, PaginationOptions } from '../interfaces';
+
+export class AutoPaginatable<T> {
+  constructor(
+    private _list: DeserializedList<T>,
```

Comment:
We don't need the leading `_` to signify that this is private, since we mark it as such with the `private` keyword:

```suggestion
    private list: DeserializedList<T>,
```

Additionally, the `_` prefix in particular will interfere with TypeScript's usage checking, as a `_` prefix is used to indicate to the compiler that a value is intentionally unused.